apps/google-analytics-4/src/locations/ConfigScreen.spec.tsx

```diff
@@ -1,32 +1,73 @@
 import React from 'react';
 import ConfigScreen from './ConfigScreen';
-import { render, RenderResult } from '@testing-library/react';
-import { mockCma, mockSdk } from '../../test/mocks';
+import { act, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { mockCma, mockSdk, validServiceKeyFile } from '../../test/mocks';
 
 jest.mock('@contentful/react-apps-toolkit', () => ({
   useSDK: () => mockSdk,
   useCMA: () => mockCma,
 }));
 
+// Helper to mock users clicking "save" -- return result of the callback passed to onConfigure()
+const saveAppInstallation = async () => {
+  // We manually call the LAST onConfigure() callback (this is important, as earlier calls have stale data)
+  return await mockSdk.app.onConfigure.mock.calls.slice(-1)[0][0]();
```

Comment:
There is now a new array method `.at(...)` to keep the code simpler: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at